lib/BannerPlugin.js

```diff
@@ -8,6 +8,10 @@
 const ConcatSource = require("webpack-sources").ConcatSource;
 const ModuleFilenameHelpers = require("./ModuleFilenameHelpers");
 
+const validateSchema = require("./validateSchema");
```

Comment:
@gribnoysup You could use [`schema-utils`](https://github.com/webpack-contrib/schema-utils) for that purpose as standardize way for plugins/loaders. If you need anything feel free to open an issue in `webpack-contrib/schema-utils`

```js
const schema = require("../schemas/plugins/bannerPluginSchema.json");
const validateOptions = require("schema-utils");

class Plugin {
  constructor (options) {
     validateOptions(schema, options, `${name}` /* <= Banner Plugin */)
  }
}
```

### `ValidationError`

![validationerror](https://user-images.githubusercontent.com/5419992/32128338-29cd12e8-bb7d-11e7-865b-87d89df12bbd.png)

Reply:
@michael-ciniawsky almost done with making requested changes, but I think I'm blocked by https://github.com/webpack-contrib/schema-utils/issues/3 at the moment 😿

I made a PR with the possible solution!

Reply:
I will try to get your PR in as soon as possible 😛  thx again